src/api_server.cc

```diff
@@ -88,6 +82,19 @@ MetadataApiServer::MetadataApiServer(const Configuration& config,
   }
 }
 
+MetadataApiServer::MetadataApiServer(const Configuration& config,
+                                     const MetadataStore& store,
+                                     int server_threads,
+                                     const std::string& host, int port)
+    : MetadataApiServer(config, store, server_threads, host, port,
+                        std::map<std::pair<std::string, std::string>, Handler>({
```

Comment:
You didn't have to make this change. First, you can test the `Dispatcher` class separately, and the `HandleMonitoredResource` callback separately (that's why the latter was pulled out in the first place). Second, the new code looks a lot uglier because of the explicit specs for the `map` and `pair` types. I would back out this whole commit.

Reply:
Looking things over now, I agree.  Ill close this pull request and open a new one testing Dispatcher directly

Reply:
Hmm. I meant this commit, not this PR. You could have reused the PR for the dispatch tests (or the `HandleMonitoredResource` test, which you still can — just reopen the PR).